doc/toggleterm.txt

```diff
@@ -466,10 +475,20 @@ You can create your on commands by using the lua functions this plugin provides
 directly
 
 >
-    command! -count=1 TermGitPush  lua require'toggleterm'.exec("git push",    <count>, 12)
-    command! -count=1 TermGitPushF lua require'toggleterm'.exec("git push -f", <count>, 12)
+    local toggleterm = require'toggleterm'
+    local strategies = toggleterm.toggle_strategies
+    local cmd = vim.api.nvim_create_user_command
+
+    cmd("TermGitPush", function(opts) toggleterm.exec("git push", opts.count, 12) end, {count = 1})
+    cmd("TermGitPushF", function(opts) toggleterm.exec("git push -f", opts.count, 12) end, {count = 1})
+    cmd("ToggleTermTab", function(opts) toggleterm.toggle_command(opts.args, strategies["by_tabpage"]()) end, {})
 <
 
+The table `require'toggleterm'.toggle_strategies` provides a set of utility
```

Comment:
I'm very, very keen to keep the API of this function very lean and quiet straightforward, I really don't want to expose a bunch of foot guns or internal APIs that then lock me into maintaining functions in a certain way. I'd much rather we focus on clear use cases rather than maybe people will find them useful apis

Reply:
The main thing here is that people (myself certainly included) may want to have the option of either smart-toggling by session or by tab, so I exposed this to allow for that. If we want to hide the strategies table altogether an alternative would be to define commands for each variant (e.g. :ToggleTermTab), I'm just not sure about the extent to which you're keen on potentially proliferating such commands variants as we get request for new smart-toggle strategies (at first glance, I personally don't have much against doing so).

Reply:
@rish987 so by default a user can add terminal to their session options which should persist all terminals in a neovim session and then toggleterm will resurrect them (think this might have regressed recently for some reason). I'm trying to ensure that the motivations for all these changes are super clear and that the code only add/fixes those specific use cases.

I'm fine with having a few strategy commands, I personally don't imagine these will grow a lot since tbh after several years of maintaining this plugin this is the first request for any alternative strategy so I don't imagine that will suddenly grow to dozens